app/services/audit/notification.rb

```diff
@@ -0,0 +1,21 @@
+module Audit
+  class Notification
+    class << self
+      include Audit::Helper
+
+      # TODO: rdoc
+      def notify(listener)
+        return unless block_given?
+
+        ActiveSupport::Notifications.instrument(instrument_name(listener), yield(Audit::Event::Payload.empty))
+      end
+
+      # TODO: rdoc
+      def listen(*args)
+        ActiveSupport::Notifications::Event.new(*args).
+          then { |event| Audit::Event::Util.serialize(event) }.
```

Comment:
What's `then`? Is it an `Event` method? I don't think I've ever seen this

Reply:
`then` is an alias for `yield_self`
https://ruby-doc.org/core-mruby/Kernel.html#method-i-then

Reply:
Wooo, didn't know either. I just knew about `tap` but I see they have different behaviors. Cool, thanks!

Reply:
Yep, `tap` is a different one. `then` and `yield_self` are relatively new. Btw, @brarnaudovski , thanks for reminding about the alias, I have managed to forget that it exists :)